From a527d43b23e2cf02993e7f899344f2959c9939f3 Mon Sep 17 00:00:00 2001
From: Brenda Streiff <brenda.streiff@emerson.com>
Date: Wed, 3 Dec 2025 16:30:36 -0600
Subject: [PATCH] Revert "lib/oe/go: document map_arch, and raise an error on
 unknown architecture"

This reverts commit e6de433ccb2784581d6c775cce97f414ef9334b1.

Returning a KeyError here breaks any go recipe at _parse time_ on an
unsupported architecture, regardless of whether that recipe would be
built for that target. This breaks microblazeel, which is necessary
for versal-based product builds.

https://lists.openembedded.org/g/openembedded-core/topic/116430143#msg226817

Given that this breaks meta-xilinx (and was complained-about by a meta-xilinx
maintainer), I expect there to be a better fix for this soon; until then,
revert the commit.

Upstream-Status: Not Submitted [immediate build fix, likely better fix upstream soon]
Signed-off-by: Brenda Streiff <brenda.streiff@emerson.com>
---
 meta/lib/oe/go.py | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/meta/lib/oe/go.py b/meta/lib/oe/go.py
index 4559dc63b2..dfd957d157 100644
--- a/meta/lib/oe/go.py
+++ b/meta/lib/oe/go.py
@@ -7,10 +7,6 @@
 import re
 
 def map_arch(a):
-    """
-    Map our architecture names to Go's GOARCH names.
-    See https://github.com/golang/go/blob/master/src/internal/syslist/syslist.go for the complete list.
-    """
     if re.match('i.86', a):
         return '386'
     elif a == 'x86_64':
@@ -35,4 +31,4 @@ def map_arch(a):
         return 'riscv64'
     elif a == 'loongarch64':
         return 'loong64'
-    raise KeyError(f"Cannot map architecture {a}")
+    return ''
-- 
2.47.3

